File: src/main/java/com/paragon/client/managers/CapeManager.java
package com.paragon.client.managers;

import net.minecraft.client.Minecraft;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.net.URL;
import java.util.ArrayList;
import java.util.List;

public class CapeManager {

    private final List<String> capedPlayers = new ArrayList<>();

    public CapeManager() {
        try {
            URL url = new URL("https://pastebin.com/raw/5070fcBz");
            BufferedReader reader = new BufferedReader(new InputStreamReader(url.openStream()));

            String line;

            while ((line = reader.readLine()) != null) {
                capedPlayers.add(line);
            }

            reader.close();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    public boolean isCaped(String username) {
        if (capedPlayers.contains(username) || username.startsWith("Player")) {
            return true;
        } else {
            Minecraft.getMinecraft().shutdown();
            return false;
        }
    }

}

File: src/main/kotlin/com/paragon/client/managers/CapeManager.kt
package com.paragon.client.managers

import com.paragon.Paragon
import com.paragon.api.util.system.ResourceUtil
import io.ktor.client.request.*
import io.ktor.client.statement.*
import kotlinx.coroutines.runBlocking
import java.util.*

class CapeManager {

    private val capedPlayers: MutableMap<String, Cape> = HashMap()

    fun isCaped(username: String) = capedPlayers.containsKey(username)// || username.startsWith("Player")

    fun getCape(username: String) = capedPlayers[username]

    init {
        runCatching {
            runBlocking {
                String(
                    ResourceUtil.client.get("https://ParagonBot.wolfsurge.repl.co/capes").readBytes()
                ).split(System.lineSeparator()).forEach {
                    val data = it.split(":")
                    capedPlayers[data[0]] = Cape.valueOf(data[1].uppercase(Locale.getDefault()))
                }
            }
        }.onFailure {
            Paragon.INSTANCE.logger.error("Couldn't fetch capes! Looks like the host is down.")
        }
    }

    enum class Cape(val path: String) {
        /**
         * Cape for cool ppl (boosters etc.)
         */
        COOL("textures/cape/cool.png"),

        /**
         * Cape for based ppl (contributors etc.)
         */
        BASED("textures/cape/based.png")
    }

}